sway/input/cursor.c

```diff
@@ -825,6 +879,20 @@ struct sway_cursor *sway_cursor_create(struct sway_seat *seat) {
 	cursor->hide_source = wl_event_loop_add_timer(server.wl_event_loop,
 			hide_notify, cursor);
 
+	cursor->pointer_gestures = wlr_pointer_gestures_v1_create(server.wl_display);
+	cursor->pinch_begin.notify = handle_pointer_pinch_begin;
+	wl_signal_add(&wlr_cursor->events.pinch_begin, &cursor->pinch_begin);
```

Comment:
These signals are not removed when the cursor is destroyed.

Reply:
Ah sorry about that. I made a change to correct this, although since the pull request was merged already I'm not sure what the best path to fixing this is.

Also on a sidenote, `wlr_pointer_gestures_v1` doesn't have a destroy or free function. It seems it handles destroying itself when the display is destroyed. Is this something worth worrying about?

Reply:
>Ah sorry about that. I made a change to correct this, although since the pull request was merged already I'm not sure what the best path to fixing this is.

If you could submit a new PR to fix this, that would be great.

>Also on a sidenote, wlr_pointer_gestures_v1 doesn't have a destroy or free function. It seems it handles destroying itself when the display is destroyed. Is this something worth worrying about?

Yeah, wlr_pointer_gestures_v1 is destroyed when the display is.